src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/samples/RESTSampleSequenceFilesController.java

```diff
@@ -526,30 +534,27 @@ public ModelMap addNewSequenceFilePairToSample(@PathVariable Long sampleId,
 		// get the sequencing run
 		SequencingRun sequencingRun = null;
 
-		if (!Objects.equal(fileResource1.getMiseqRunId(), fileResource2.getMiseqRunId())) {
-			throw new IllegalArgumentException("Cannot upload a pair of files from different sequencing runs");
-		}
+		try {
```

Comment:
Same with in this method.  Move as much of the file stuff like `file1.transferTo` into the try as you can so that the `finally` can clean up the tmp files.

Reply:
Did the same here as the other block